sentry-core/src/main/java/io/sentry/core/SentryEnvelopeItemHeader.java

```diff
@@ -1,16 +1,25 @@
 package io.sentry.core;
 
-final class SentryEnvelopeItemHeader {
+import org.jetbrains.annotations.Nullable;
+
+import java.util.concurrent.Callable;
+
+public final class SentryEnvelopeItemHeader {
   private final String contentType;
   private final String fileName;
   private final String type;
   private final int length;
+  @Nullable private final Callable<Integer> getLength;
 
+  // TODO: Looks like a type here that defaults to String for unknown values would be ideal
   public String getType() {
     return type;
   }
 
   public int getLength() {
+    if (getLength != null) {
+      return getLength();
+    }
```

Comment:
let's rename `getLength` field, it looks an endless loop method

Reply:
might be since callables need to be invoked with `.call()`